docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,1975 @@
+
+[discrete]
+=== Before you begin
+
+Create a deployment using our hosted {ess} on https://www.elastic.co/cloud/[Elastic Cloud]. The deployment includes
+an {es} cluster for storing and searching your data, {kib} for visualizing and managing
+your data, and an APM server. For more information, see <<set-up-on-cloud,Set up on Cloud>>.
+
+[discrete]
+=== Step 1: Create a Java application
+
+To create the Java application, you require OpenJDK 14 (or higher) and the https://javalin.io/[Javalin]
+web framework. The application will include the main endpoint, an
+artificially long-running endpoint, and an endpoint that needs to poll
+another data source. Finally, there will be a background job
+running. 
+
+. Set up a Gradle project and create the following `build.gradle` file.
++
+[source,gradle]
+----
+plugins {
+  id 'java'
+  id 'application'
+}
+
+repositories {
+  jcenter()
+}
+
+dependencies {
+  implementation 'io.javalin:javalin:3.9.1'
+
+  testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
+  testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.6.2'
+}
+
+application {
+  mainClassName = 'de.spinscale.javalin.App'
+}
+
+test {
+  useJUnitPlatform()
+}
+----
++
+. Run the following command.
++
+[source,bash]
+----
+echo "rootProject.name = 'javalin-app'" >> settings.gradle
+
+mkdir -p src/main/java/de/spinscale/javalin
+mkdir -p src/test/java/de/spinscale/javalin
+----
++
+. Install the Gradle wrapper. An easy way to install Gradle is to use
+https://sdkman.io/[sdkman] and run `sdk install gradle 6.5.1`.
++
+. Run `./gradlew clean check`. You should see a
+successful build that has nothing built or compiled yet.
++
+. To create a Javalin server and its first endpoint (the main endpoint), create the
+`src/main/java/de/spinscale/javalin/App.java` file.
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.javalin.Javalin;
+
+public class App {
+    public static void main(String[] args) {
+        Javalin app = Javalin.create().start(7000);
+        app.get("/", ctx -> ctx.result("Appsolutely perfect"));
+    }
+}
+----
++
+. Run `./gradlew assemble`
++
+This creates a compiled `App.class` file in the `build` directory.
+However, there is no way to start the server. Let’s create a jar which
+contains our compiled class along with all the required dependencies.
++
+. In the `build.gradle` file, edit `plugins` as shown here.
++
+[source,gradle]
+----
+plugins {
+  id 'com.github.johnrengelman.shadow' version '6.0.0'
+  id 'java'
+}
+----
++
+. Run `./gradlew shadowJar`. This creates a file
+`build/libs/javalin-app-all.jar`, which now can be run as shown here.
++
+[source,bash]
+----
+java -jar build/libs/javalin-app-all.jar
+----
++
+The `shadowJar` plugin requires some information about its main class. Add the
+following snippet to the `build.gradle` file.
++
+....
+jar {
+  manifest {
+    attributes 'Main-Class': 'de.spinscale.javalin.App'
+  }
+}
+....
```

Comment:
There are a few code blocks like this that aren't formatted correctly. I'm guessing you just haven't got around to this part of the review yet.

Reply:
Thanks. Updated the code blocks.